e2e/benchmarks/local-benchmark/dump.js

```diff
@@ -0,0 +1,235 @@
+      // This log informs tools file has been saved.
+      console.log(fileName);
+    }
+  }
+}
+
+/**
+ * Create a NamedTensorMap from an output node name.
+ * @param outputNodeName output node name.
+ * @param modelJson The parsed model.json.
+ * @param dumpedJson The dumped tensor infomation (including shape, dtype,
+ *     value).
+ *
+ * @returns A NamedTensorMap.
+ */
+async function createTensorMap(outputNodeName, modelJson, dumpedJson) {
+  const modelNodes = modelJson['modelTopology']['node'];
+  let inputs = [];
+  for (let i = 0; i < modelNodes.length; i++) {
+    if (outputNodeName === modelNodes[i].name && modelNodes[i].input) {
+      inputs = modelNodes[i].input;
+      break;
+    }
+  }
+  // In
+  // https://storage.googleapis.com/tfhub-tfjs-modules/mediapipe/tfjs-model/face_landmarks_detection/attention_mesh/1/model.json,
+  // some inputs are prefixed with '^'.
+  if (!inputs || inputs.length == 0 || inputs[0].startsWith('^')) {
+    return null;
+  }
+
+  let tensorMap = {};
+  for (let i = 0; i < inputs.length; i++) {
+    const key = inputs[i].split(':')[0];
+    if (dumpedJson[key] == null || dumpedJson[key][0] == null) {
+      console.warn('Tensor ' + key + ' is null!');
+      return null;
+    }
+    const tensorInfo = dumpedJson[key][0];
+    const tensor = tf.tensor(
+        Object.values(tensorInfo.value[0]), tensorInfo.shape, tensorInfo.dtype);
+    tensorMap[key] = tensor;
+  }
+
+  return tensorMap;
+}
+
+async function predictOp(
+    model, modelJson, dumpedJson, outputNodeName, backend) {
+  await tf.setBackend(backend);
+  const tensorMap =
+      await createTensorMap(outputNodeName, modelJson, dumpedJson);
+  if (tensorMap == null) {
+    return null;
+  }
+  let prediction;
+  try {
+    // TODO(#6861): Support tensor with type conversion.
+    prediction = await model.executeAsync(tensorMap, outputNodeName);
+  } catch (e) {
+    console.warn(e.message);
+    return null;
+  }
+
+  let intermediateData =
+      await getIntermediateTensorInfo(model.getIntermediateTensors());
+  await getPredictionData(prediction);
+  model.disposeIntermediateTensors();
+  return intermediateData;
+}
+
+async function dumpOp(model, dumpedJson, backends, outputNodeName, index) {
+  let objects1 = {};
+  let objects2 = {};
+  const modelJson = model.artifacts;
+  const referenceObject = await predictOp(
+      model, modelJson, dumpedJson, outputNodeName, backends[1]);
+
+  const predictObject = await predictOp(
+      model, modelJson, dumpedJson, outputNodeName, backends[0]);
+  if (predictObject && referenceObject) {
+    objects1 = {index, ...predictObject[`${outputNodeName}`]};
+    objects2 = {index, ...referenceObject[`${outputNodeName}`]};
+  }
+  return [objects1, objects2];
+}
+
+
+/**
+ * Dump the predict results of two backends and save diffs to files.
+ * @param model The loaded model.
+ * @param jsonObjects The predict results from backends.
+ * @param backends [predict backend, reference backend].
```

Comment:
jsonObjects is not meaningful. Can we combine these two (jsonObjects and backends) as {backend0: intermediateTensor0, backend1: intermediatetensor1}?

Reply:
Done